README.md

```diff
@@ -2,31 +2,189 @@
+![hit-breakpoint-correct-name](https://curriculum-content.s3.amazonaws.com/spring-mod-1/debugging/debug-name-ted.png)
 
-## Resources
+We'll now see that the `name` variable is accurately set to `Ted` now. If we
+continue running the application, we'll see that it returns `Hello Ted!` now.
+
+## Other Debugging Tips
+
+In the example above, we saw how to debug a bug that might not have been very
+obvious.
+
+Let's look at another example of how to use the debugger. This time, we'll refer
+back to the football team example we have been using.
+
+Maybe we want to look at the request body that was passed into the application.
+We can set a breakpoint to do so!
+
+![football-controller-breakpoint](https://curriculum-content.s3.amazonaws.com/spring-mod-1/debugging/debug-football-breakpoint.png)
+
+When we hit the breakpoint, we can look in the debug console window to see
+exactly what was sent and if it matches what we expected:
+
+![football-controller-console](https://curriculum-content.s3.amazonaws.com/spring-mod-1/debugging/debug-football-console.png)
+
+Other common issues include:
+
+- The path variable names are misspelled or different from each other.
+
+![misspelled-path-variable](https://curriculum-content.s3.amazonaws.com/spring-mod-1/debugging/debug-misspelled-path-variable.png)
+
+This could result in a warning:
+
+```text
+WARN 25306 --- [nio-8080-exec-4] .w.s.m.s.DefaultHandlerExceptionResolver : Resolved [org.springframework.web.bind.MissingPathVariableException: Required URI template variable 'names' for method parameter type String is not present]
+```
+
+- Using the wrong annotations to perform a request mapping.
+
+![wrong-annotation](https://curriculum-content.s3.amazonaws.com/spring-mod-1/debugging/debug-wrong-annotation.png)
+
+This could result in a warning:
+
+```text
+WARN 28814 --- [nio-8080-exec-2] .w.s.m.s.DefaultHandlerExceptionResolver : Resolved [org.springframework.web.HttpRequestMethodNotSupportedException: Request method 'GET' not supported]
+```
+
+- Using a `@RequestParam` instead of a `@PathVariable` and vice versa.
+
+![requestparam-vs-pathvariable](https://curriculum-content.s3.amazonaws.com/spring-mod-1/debugging/debug-requestparam-instead-pathvariable.png)
+
+This could result in a warning:
+
+```text
+WARN 29881 --- [nio-8080-exec-1] .w.s.m.s.DefaultHandlerExceptionResolver : Resolved [org.springframework.web.bind.MissingServletRequestParameterException: Required request parameter 'name' for method parameter type String is not present]
+```
+
+## Conclusion
 
-- [Resource Link 1](example.com)
-- [Resource Link 2](example.com)
+Seeing what is sent to the application and what it is getting back from the
+data source is incredibly helpful when trying to figure out the unexpected. The
+debugger tool in IntelliJ is not to be overlooked and can serve as an
+efficient and quick way to figure out what may be wrong. Using this in
+conjunction with Postman and the Java Visualizer tool will help in debugging
```

Comment:
Do you still mention the Java Visualizer elsewhere?

Reply:
I added a clarification that, even though the Java Visualizer wasn't used here, students should still be encouraged to use it when appropriate if it helps. Assumption is that some of the lessons from previous modules will include how to use the Java Visualizer in more detail.